presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -152,6 +154,7 @@ public static ParquetPageSource createParquetPageSource(
             TupleDomain<HiveColumnHandle> effectivePredicate,
             FileFormatDataSourceStats stats)
     {
+        checkState(columns.stream().allMatch(column -> column.getColumnType() == REGULAR), "column type must be regular: %s", columns);
```

Comment:
This should be `checkArgument` since this is validating input arguments. I think it'd be cleaner to do a normal loop and only show the one that failed:
```java
for (HiveColumnHandle column : columns) {
    checkArgument(column.getColumnType() == REGULAR, "column type must be REGULAR: %s", column);
}